fs/dafsclient/rmtclient.cpp

```diff
@@ -1085,81 +1097,122 @@ ISocket *checkSocketSecure(ISocket *socket)
     return LINK(socket);
 }
 
-ISocket *connectDafs(SocketEndpoint &ep, unsigned timeoutms)
+ISocket *connectDafs(SocketEndpoint &ep, unsigned timeoutms, const IPropertyTree *service)
 {
     Owned<ISocket> socket;
 
-    if ( (securitySettings.queryConnectMethod() == SSLNone) || (securitySettings.queryConnectMethod() == SSLOnly) )
+    if (isContainerized())
     {
         socket.setown(ISocket::connect_timeout(ep, timeoutms));
-        return checkSocketSecure(socket);
-    }
-
-    // SSLFirst or UnsecureFirst ...
-
-    unsigned newtimeout = timeoutms;
-    if (newtimeout > 5000)
-        newtimeout = 5000;
 
-    int conAttempts = 2;
-    while (conAttempts > 0)
-    {
-        conAttempts--;
-        bool connected = false;
-        try
+        if (service->getPropBool("@tls"))
```

Comment:
Can service be null when containerized?  I see several places passing nullptr.. perhaps should throw an exception if null?

Reply:
That question highlighted one place where it should potentially also have passed the service (in RemotePartitioner).
However, I'm not entirely sure how or even if, that route can be taken in containerized mode, and it would likely not have worked before anyway, because the legacy ftslave route was also incomplete.

Nevertheless for completeness, I'll modify that code, which will mean commoning up the code that gets the service.

The other places that call connectDaFs that pass null, are bare-metal only and cases that I don't think will ever be used in directly containerized mode (mostly dafscontrol). Rather than throw an exception.
I will change the if to be conditional on 'service' and let the code continue to try without certificates.